File: e-elearning-main/h-learning-fe-admin/src/model/video_lession.ts
import { BaseModel } from "./base";
import { LessionModel } from "./lession";

export type VideoLessionModel = BaseModel & {
    code: string
    thumnail: string
    url360p?: string
    url480p?: string
    url720p?: string
    url1080p?: string
    status?: VIDEO_LESSION_STATUS
    lessionId: number

    lession?: LessionModel
}

export type VIDEO_LESSION_STATUS = 
  |"pending" 
  | "done";

File: e-elearning-main/service/internal/entity/video_lession.go
package entity

import "gorm.io/gorm"

type VideoLession struct {
	gorm.Model
	Code     string                `json:"code" gorm:"unique"`
	Thumnail string                `json:"thumnail"`
	Url360p  *string               `json:"url360p"`
	Url480p  *string               `json:"url480p"`
	Url720p  *string               `json:"url720p"`
	Url1080p *string               `json:"url1080p"`
	Status   *VIDEO_LESSION_STATUS `json:"status"`

	LessionId uint     `json:"lessionId"`
	Lession   *Lession `json:"lession" gorm:"foreignKey:LessionId; constraint:OnUpdate:CASCADE,OnDelete:SET NULL;"`
}

type VIDEO_LESSION_STATUS string

const (
	VIDEO_LESSION_PENDING VIDEO_LESSION_STATUS = "VIDEO_LESSION_PENDING"
	VIDEO_LESSION_DONE    VIDEO_LESSION_STATUS = "VIDEO_LESSION_DONE"
)
